homeassistant/components/rachio/binary_sensor.py

```diff
@@ -140,3 +151,26 @@ async def async_added_to_hass(self) -> None:
                 self._async_handle_any_update,
             )
         )
+
+
+class RachioHoseTimerBattery(RachioHoseTimerEntity, BinarySensorEntity):
+    """Represents a battery sensor for a smart hose timer."""
+
+    _attr_device_class = BinarySensorDeviceClass.BATTERY
+
+    def __init__(
+        self, data: dict[str, Any], coordinator: RachioUpdateCoordinator
+    ) -> None:
+        """Initialize a smart hose timer battery sensor."""
+        super().__init__(data, coordinator)
+        self._attr_unique_id = f"{self.id}-battery"
+        self._static_attrs = data[KEY_STATE][KEY_REPORTED_STATE]
+        self._attr_is_on = self._static_attrs[KEY_BATTERY_STATUS] == KEY_LOW
```

Comment:
```suggestion
        self._update_attr()
```

Untested, but I think this will work

Reply:
👍 I'll give it a try

Reply:
That works, probably could make that same change to the switch as well

Reply:
In theory that line could be moved out of binary sensor and switch to the base class constructor in `entity.py`, let me try that.